### DISCUSSION TITLE
openSUSE Dependencies

### DISCUSSION THREAD

**Discussion by Espionage724** [alvr-org/ALVR#709] [Show and tell]
Recently got ALVR up and running on openSUSE Tumbleweed, and figure someone may find the list of dependencies useful.

Here's what I have after some trial-and-error (some may be unnecessary):

```
sudo zypper install gcc-c++ gcc libopenssl-devel libunwind-devel yasm nasm curl libx264-devel libx265-devel vulkan-devel ffmpeg-4-libavcodec-devel ffmpeg-4-libavutil-devel libdrm-devel make glib2-devel cairo-devel pango-devel gdk-pixbuf-devel atk-devel gtk3-devel clang-devel libva-devel valgrind-devel
```

That will get ALVR (https://github.com/alvr-org/ALVR/commit/813b8b792289c093c5c3495ede9b1467e0e549c7) compiled on openSUSE Tumbleweed with `cargo xtask build-server --release --bundle-ffmpeg`.

According to one of the ALVR logs, a Chromium install is checked for and an error is presented in the log when one isn't found (`alvr/server/src/lib.rs:180`), but it doesn't seem required for general functionality (settings can be adjusted from http://localhost:8082/ with Firefox)

HEVC didn't originally work for me (RX 580), but https://github.com/alvr-org/ALVR/issues/706 mentioned needing `valgrind`, which seems to have also fixed it in my case.

**Comment by Alfenstein8** [comment]
Also
```
sudo zypper install alsa-devel
```
